OpenRA.Game/Network/ReplayConnection.cs

```diff
@@ -81,6 +90,25 @@ public ReplayConnection(string replayFilename)
 						TickCount = Math.Max(TickCount, frame);
 					}
 				}
+
+				// 2nd parse : replace all disconnect packets without frame with real
+				// disconnect frame
+				// NOTE: to modify/remove if a reconnect feature is set
+				foreach (var tmpChunk in chunks)
+				{
+					foreach (var tmpPacketPair in tmpChunk.Packets)
+					{
+						var client = tmpPacketPair.First;
+						var packet = tmpPacketPair.Second;
+						if (packet.Length == 5 && packet[4] == 0xBF)
+						{
+							int lastClientFrame = lastClientsFrame[client];
+							byte[] lastFramePacket = BitConverter.GetBytes(lastClientFrame);
+							if (lastFramePacket.Length == 4)
```

Comment:
It was to be sure that lastFramePacket always be 4, no more no less. Otherwise replacing a arbitrary value to the packet can cause crash. 
But totally miss-viewed that BitConverter.GetBytes(int32) return every time a 4 byte array.